fix: add loading indicator to UserForm submit button

harsh04044 opened the pull request:
Fixes #14837 

## Proposed Changes

This PR adds a loading indicator to the UserForm submit button to provide better visual feedback during user creation/update. Previously, the button would just become disabled without any indication that the form was being submitted, which could confuse users.

- Add loading spinner icon and "Saving..." text when the form is submitting (using `createPending` or `updatePending` states)
- Disable the submit button when form has validation errors to prevent showing loading state unnecessarily
- Follows the same pattern used in other forms like ProductKnowledgeForm for consistency

The button now shows a spinner and changes text to "Saving..." when submitting, and stays disabled when there are validation errors (like invalid email format) so users get immediate feedback without triggering a submission.

## Before
<img width="1470" height="956" alt="image" src="https://github.com/user-attachments/assets/05fe9ab5-7e47-499f-8b1f-457cfb3aa987" />


## After
<img width="2940" height="1912" alt="image" src="https://github.com/user-attachments/assets/174a71ae-48f5-4186-a8c0-2ce4e0c91ce2" />


Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is placed in I18n files.
- [ ] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [ ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.
- [ ] Add or update Playwright tests for related changes

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Submit button now shows a loading state with context-specific labels ("creating" or "updating") while an operation is pending for clearer feedback.
  * Button remains disabled during pending operations and when the form is incomplete; when idle, it preserves distinct "create" vs "update" labels.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

harsh04044 committed:
fix: add loading indicator to UserForm submit button

- Add loading spinner and 'Saving...' text when form is submitting
- Disable button when form has validation errors to prevent unnecessary loading state
- Improve UX by providing clear visual feedback during submission

Fixes #14837

AdityaJ2305 commented on src/components/Users/UserForm.tsx:
```diff
@@ -826,10 +826,24 @@ export default function UserForm({
             updatePending ||
-            createPending
+            createPending ||
+            !form.formState.isValid
```
Let’s allow the user to submit the form; otherwise, they won’t know exactly what error they’re making

harsh04044 replied:
okay thats right , i have changed it in my latest commit

harsh04044 committed:
refactor: remove isValid check from UserForm submit button

Allow users to submit form to see validation errors instead of disabling button when form is invalid

AdityaJ2305 commented on src/components/Users/UserForm.tsx:
```diff
@@ -829,7 +829,20 @@ export default function UserForm({
+            <>
+              <CareIcon
+                icon="l-spinner"
+                className="mr-2 h-4 w-4 animate-spin"
```
```suggestion
                className="mr-2 size-4 animate-spin"
```

harsh04044 committed:
Update src/components/Users/UserForm.tsx

Co-authored-by: Aditya Jindal <adityajindal2305@gmail.com>

harsh04044 committed:
fix: format CareIcon component to single line

Fixes Prettier formatting error introduced by commit suggestion

nihal467 commented on src/components/Users/UserForm.tsx:
```diff
@@ -829,7 +829,17 @@ export default function UserForm({
+            <>
+              <CareIcon icon="l-spinner" className="mr-2 size-4 animate-spin" />
+
+              {t("saving")}
```
- text needs to be based on isEditMode
- remove the spinner

harsh04044 replied:
@nihal467  i have addresed your review in the latest commit.

nihal467 reviewed (changes requested):
.

harsh04044 committed:
fix: remove spinner and make button text conditional based on isEditMode

- Remove CareIcon spinner from submit button
- Replace static 'saving' text with conditional text based on isEditMode
- Show 'updating'/'creating' when pending, 'update_user'/'create_user' when not pending
- Addresses maintainer feedback from PR #14861

harsh04044 commented:
**About the failing Tests**
The CI failures are unrelated to this PR. They're in patient age button selectors and questionnaire visibility tests, not the UserForm submit button changes.
This  latest commit only removes the spinner and makes the button text conditional based on isEditMode. No UserForm tests failed, and the user creation test still expects "Create User", which matches our implementation.
These look like pre-existing flaky tests.

nihal467 committed:
Merge branch 'develop' into fix/userform-loading-indicator-main
